dhis-2/dhis-api/src/main/java/org/hisp/dhis/i18n/I18nFormat.java

```diff
@@ -222,9 +223,10 @@ public String formatPeriod( Period period )
 
         String typeName = period.getPeriodType().getName();
 
-        if ( typeName.equals( WeeklyPeriodType.NAME ) ) // Use ISO dates due to potential week confusion
+        if ( typeName.contains( WeeklyPeriodType.NAME ) ) // Use ISO dates due to potential week confusion
         {
-            return period.getIsoDate();
+            DateTime dateTime = new DateTime( period.getStartDate() );
+            return "W" + dateTime.weekOfWeekyear().getAsText() + " " + (typeName.equals( WeeklyPeriodType.NAME ) ? dateTime.year().getAsText() : dateTime.dayOfWeek().getAsShortText() + " " + dateTime.year().getAsText());

```

Comment:
This one looks a bit hacky. Can we use the WeeklyPeriodType object or some other existing class to convert to the correct period name?